go.mod

```diff
@@ -3,11 +3,12 @@ module github.com/containerd/go-cni
 go 1.21
 
 require (
-	github.com/containernetworking/cni v1.1.2
+	github.com/containernetworking/cni v1.2.0
 	github.com/stretchr/testify v1.8.0
 )
 
 require (
+	github.com/Masterminds/semver/v3 v3.2.1 // indirect
```

Comment:
background discussion with CNI team if exposure of this package was on purpose .. check if a newer tagged version of this semver project is needed etc..

Reply:
@squeed thoughts? We could just revert that pr unless we have a better route

Reply:
Noting nil version equality fix for panic.. https://github.com/Masterminds/semver/pull/213 is not in this tagged version. Would prefer using golang public `[]semver` or `[]string` if possible to reduce externalized package dependencies. 

That said we can handle the two asks separately if desired by the CNI team.